crates/bevy_core_pipeline/src/auto_exposure/compensation_curve.rs

```diff
@@ -0,0 +1,212 @@
+    /// let curve: Handle<AutoExposureCompensationCurve> = compensation_curves.add(
+    ///    LinearSpline::new([
+    ///        vec2(-4.0, -2.0),
+    ///        vec2(0.0, 0.0),
+    ///        vec2(2.0, 0.0),
+    ///        vec2(4.0, 2.0),
+    ///    ])
+    ///    .to_curve()
+    /// );
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// Panics if the end of a segment is not equal to the beginning of the next segment,
+    /// or if the curve is not monotonically increasing on the x-axis.
+    fn from(curve: CubicCurve<Vec2>) -> Self {
+        let min_log_lum = curve.position(0.0).x;
+        let max_log_lum = curve.position(curve.segments().len() as f32).x;
+        let domain = max_log_lum - min_log_lum;
+        let step = domain / 255.0;
+        assert!(min_log_lum < max_log_lum);
+
+        let mut data = [0.0; 256];
+        let mut min_compensation = std::f32::MAX;
+        let mut max_compensation = std::f32::MIN;
+        let mut previous_end = min_log_lum;
+
+        for segment in curve {
+            let begin = segment.position(0.0).x;
+            let end = segment.position(1.0).x;
+            assert!(begin < end);
+            assert!(begin == previous_end);
+            previous_end = end;
+
+            let i = (((begin - min_log_lum) / domain) * 255.0).ceil() as usize;
+            let j = (((end - min_log_lum) / domain) * 255.0).floor() as usize;
+
+            for (k, v) in data[i..=j].iter_mut().enumerate() {
+                *v = find_y_given_x(&segment, min_log_lum + (i + k) as f32 * step);
+                min_compensation = v.min(min_compensation);
+                max_compensation = v.max(max_compensation);
+            }
+        }
+
+        let compensation_range = max_compensation - min_compensation;
+
+        Self {
+            min_log_lum,
+            max_log_lum,
+            min_compensation,
+            max_compensation,
+            data: if compensation_range > 0.0 {
+                data.map(|f: f32| {
+                    (((f - min_compensation) / compensation_range) * 255.0).clamp(0.0, 255.0) as u8
+                })
+            } else {
+                [0; 256]
+            },
+        }
+    }
+}
+
+/// Maximum allowable error for iterative Bezier solve
+const MAX_ERROR: f32 = 1e-5;
+
+/// Maximum number of iterations during Bezier solve
+const MAX_ITERS: u8 = 8;
+
+/// Find the `y` value of the curve at the given `x` value using the Newton-Raphson method.
+#[inline]
+fn find_y_given_x(segment: &CubicSegment<Vec2>, x: f32) -> f32 {
+    let mut t_guess = x;
+    let mut pos_guess = Vec2::ZERO;
+    for _ in 0..MAX_ITERS {
+        pos_guess = segment.position(t_guess);
+        let error = pos_guess.x - x;
+        if error.abs() <= MAX_ERROR {
+            break;
+        }
+        // Using Newton's method, use the tangent line to estimate a better guess value.
+        let slope = segment.velocity(t_guess).x; // dx/dt
+        t_guess -= error / slope;
+    }
```

Comment:
I modified this part and took a bit of a different approach to converting the curve to a texture: the `CubicSegment`s are sampled 64 times and the resulting lines are simply plotted to the texture (in 1D fashion). Much easier to check for invalid curves this way, so it's more robust. If something is wrong with the curve an error is returned instead of a panic.